src/main/java/uk/gov/pay/adminusers/service/ServiceOtpDispatcher.java

```diff
@@ -15,20 +17,33 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(ServiceOtpDispatcher.class);
     private final InviteDao inviteDao;
     private final SecondFactorAuthenticator secondFactorAuthenticator;
+    private final PasswordHasher passwordHasher;
     private final NotificationService notificationService;
 
     @Inject
-    public ServiceOtpDispatcher(InviteDao inviteDao, SecondFactorAuthenticator secondFactorAuthenticator, NotificationService notificationService) {
+    public ServiceOtpDispatcher(InviteDao inviteDao, SecondFactorAuthenticator secondFactorAuthenticator,
+                                PasswordHasher passwordHasher, NotificationService notificationService) {
         super();
         this.inviteDao = inviteDao;
         this.secondFactorAuthenticator = secondFactorAuthenticator;
+        this.passwordHasher = passwordHasher;
         this.notificationService = notificationService;
     }
 
     @Override
     public boolean dispatchOtp(String inviteCode) {
         return inviteDao.findByCode(inviteCode)
                 .map(inviteEntity -> {
+                    Optional.ofNullable(inviteOtpRequest.getTelephoneNumber())
+                            .map(TelephoneNumberUtility::formatToE164)
+                            .ifPresent(inviteEntity::setTelephoneNumber);
+
+                    Optional.ofNullable(inviteOtpRequest.getPassword())
+                            .map(passwordHasher::hash)
+                            .ifPresent(inviteEntity::setPassword);
+
+                    inviteDao.merge(inviteEntity);
+
                     int newPassCode = secondFactorAuthenticator.newPassCode(inviteEntity.getOtpKey());
```

Comment:
might be worth extracting this whole lambda? It's getting a bit straggly. Happy for that to be in follow up, or for you to tell me otherwise...